docs/src/components/MarkdownContent.astro

```diff
@@ -7,98 +7,37 @@ import DefaultMarkdownContent from '@astrojs/starlight/components/MarkdownConten
 </DefaultMarkdownContent>
 
 <script>
-  const SvgNs = 'http://www.w3.org/2000/svg'
-
-  const copyButtonTemplate = document.createElement('button')
-  copyButtonTemplate.className = 'code-block-copy'
-  const copyButtonSvg = document.createElementNS(SvgNs, 'svg')
-  copyButtonSvg.setAttribute('viewBox', '0 0 24 24')
-  copyButtonSvg.innerHTML =
-    '<path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path>'
-  copyButtonTemplate.appendChild(copyButtonSvg)
-
-  const codeBlocks = document.querySelectorAll(
-    'pre.astro-code',
-  ) as NodeListOf<HTMLElement>
-  for (const codeBlock of codeBlocks) {
-    const codeText = codeBlock.innerText
-
-    const copy = async () => {
-      try {
-        await navigator.clipboard.writeText(codeText)
-      } catch (error) {
-        console.error(error)
-        return
-      }
-
-      copyButton.classList.add('highlight')
-      setTimeout(() => copyButton.classList.remove('highlight'), 500)
-    }
-
-    codeBlock.onscroll = () => {
-      const scroll = codeBlock.scrollLeft
-      requestAnimationFrame(() => {
-        codeBlock.style.setProperty('--scroll', scroll + 'px')
+  let _mermaidApi: import('mermaid').Mermaid
```

Comment:
Just put it into separate module and you will be able to import in as a normal singleton. Astro should allow you to import local files as a modules even from inline scripts